### ISSUE TITLE
Config files needed for customizing dmenu

### ISSUE THREAD

**Issue by ghost** [Shizcow/dmenu-rs#42]
Please make the dmenu customizable using config files in ~/.config/dmenu

Preferably toml or yml file.

**Comment by Shizcow** [comment]
I've thought about this one a lot. The only way I'd implement config files is if I also revamped the plug-in system. I want to do this, but won't have time in my life for quite a while.

Some ideas/desired features on this topic:
- ability to set default font / colors / etc
- ability to add default parameters to the command (as a fallback)
- ability to modify top / bottom / other __already__ supported UI options

As for how plugins can interact:
- remove the compile time step, allowing this to be shipped with `cargo install`
- have plugins look in 3 locations during startup:
  - system package / run path. This way, in theory, you could install dmenu-rs plugins via your system package manager. Same as how arch handles python packages 
  - .cargo: so plugins could be installed via `cargo install`
  - .config/dmenu: manual installation and plugins in development
- Alternatively / in addition to, plugins may need to be registered before use so they're not automatically loaded in (speed?)
- plugins can provide toml configuration keys available to modify in the main config file


Other ideas welcome

**Comment by Shizcow** [comment]
New plugin system won't ever happen without https://github.com/rust-lang/rfcs/issues/1880. I've been waiting on this for a few years and it doesn't look like it's going to happen any time soon. So unfortunately closing this as wontfix.
